ext/libxml/libxml.c

```diff
@@ -851,7 +839,11 @@ static PHP_RINIT_FUNCTION(libxml)
 
 static PHP_RSHUTDOWN_FUNCTION(libxml)
 {
-	_php_libxml_destroy_fci(&LIBXML(entity_loader).fci, &LIBXML(entity_loader).object);
+	LIBXML(entity_loader).fci.size = 0;
+	if (!Z_ISNULL(LIBXML(entity_loader).callback)) {
```

Comment:
AFAIK you should also cleanup the callback when it *refers to null*, since the zval itself is not null.

Reply:
Done. I don't think it's necessary, since zval_ptr_dtor_nogc() is a no-op on when the type is null. Indeed the zval* is not null, it is in thread-local storage. It can't be freed. It's the underlying array or object that I'm freeing here.